src/ai/backend/agent/server.py

```diff
@@ -633,6 +640,7 @@ async def server_main(
     async with agent:
         stop_signal = yield
         agent.mark_stop_signal(stop_signal)
+        m.close()
```

Comment:
Let's use `contextlib.closing` to ensure closing of the monitor regardless what happens in the middle.
```suggestion
    with closing(monitor):
        async with agent:
            stop_signal = yield
            agent.mark_stop_signal(stop_signal)
```
or, equivalently,
```suggestion
    try:
        async with agent:
            stop_signal = yield
            agent.mark_stop_signal(stop_signal)
    finally:
        monitor.close()
```